src/modules/Tasklist.js

```diff
@@ -178,36 +178,33 @@ export default class Tasklist {
   };
 
   #clearcompleted = () => {
-    let changed = false;
-    for (let i = 0; i < this.tasks.length; i += 1) {
-      const { completed } = this.tasks[i];
-      if (completed) {
-        this.tasks.splice(i, 1);
-        i -= 1;
-        changed = true;
-      }
-    }
-    if (changed) {
-      this.savedata();
-      this.#clearlist();
-      this.renderdata();
-    }
+    const result = this.tasks.filter((task) => task.completed === false);
+    this.tasks = [...result];
+    this.savedata();
+    this.#clearlist();
+    this.renderdata();
```

Comment:
✅ A function for the "Clear all completed" button using the `Array.filter()` method is implemented 👍🏻.